numpy/_core/src/umath/clip.cpp

```diff
@@ -196,6 +196,69 @@ _npy_clip(char **args, npy_intp const *dimensions, npy_intp const *steps)
     return _npy_clip_<Tag>((T **)args, dimensions, steps);
 }
 
+template <class Tag, class T = typename Tag::type>
+static void
+_npy_clip_floating_(T **args, npy_intp const *dimensions, npy_intp const *steps)
+{
+    npy_intp n = dimensions[0];
+    if (steps[1] == 0 && steps[2] == 0) {
+        /* min and max are constant throughout the loop, the most common case
+         */
+        T min_val = *args[1];
+        T max_val = *args[2];
+
+        T *ip1 = args[0], *op1 = args[3];
+        npy_intp is1 = steps[0] / sizeof(T), os1 = steps[3] / sizeof(T);
+
+        if (!npy_isnan(min_val) && !npy_isnan(max_val)) {
+            /* min_val and max_val are not nans */
+            /* nans in the input will be propagated naturally */
+
+            /* contiguous, branch to let the compiler optimize */
+            if (is1 == 1 && os1 == 1) {
+                for (npy_intp i = 0; i < n; i++, ip1++, op1++) {
+                    T x = *ip1;
+                    if (x < min_val) x = min_val;
+                    if (x > max_val) x = max_val;
+                    *op1 = x;
+                }
+            }
+            else {
+                for (npy_intp i = 0; i < n; i++, ip1 += is1, op1 += os1) {
+                    T x = *ip1;
+                    if (x < min_val) x = min_val;
+                    if (x > max_val) x = max_val;
```

Comment:
```suggestion                    
                    if (x > max_val) x = max_val;
                    else if (x < min_val) x = min_val;
```
(not sure whether this is really faster or more readable, but maybe worth a try)

Reply:
https://godbolt.org/z/MTxasfG33
The version I provided has better generated code. (3 instructions and branchless)

Reply:
Nice tool! Is indeed better